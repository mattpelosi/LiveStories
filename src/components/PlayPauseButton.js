import React from "react";
import PlayArrowIcon from "@material-ui/icons/PlayArrow";
import PauseIcon from "@material-ui/icons/Pause";
import Button from "@material-ui/core/Button";
import { withStyles } from "@material-ui/core/styles";

const styles = theme => ({
  button: {
    margin: theme.spacing.unit,
    color: "#3F51B5"
  }
});

function PlayPauseButton(props) {
  const { classes } = props;
  return (
    <Button
      size="large"
      className={classes.button}
      onClick={props.onClick}
      variant="outlined"
      data-test="button"
    >
      {props.paused ? <PlayArrowIcon /> : <PauseIcon />}
    </Button>
  );
}

export default withStyles(styles)(PlayPauseButton);
