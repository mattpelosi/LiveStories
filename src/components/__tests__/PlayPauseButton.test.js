import React from "react";
import PlayPauseButton from "../PlayPauseButton";
import { shallow, render, mount } from "enzyme";

describe("Conditional rendering and onClick event", () => {
  test("renders play icon when props.paused = true", () => {
    const wrapper = render(<PlayPauseButton paused={true} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("renders pause icon when props.paused = false", () => {
    const wrapper = render(<PlayPauseButton paused={false} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Onclick triggers when user clicks the button", () => {
    const onButtonClickMock = jest.fn();
    const wrapper = shallow(<PlayPauseButton onClick={onButtonClickMock} />);
    wrapper.simulate("click");
    expect(onButtonClickMock).toHaveBeenCalledTimes(1);
  });

  test("renders play icon when props.paused = true", () => {
    const wrapper = mount(<PlayPauseButton />);
    // const button = wrapper.find('[data-test="button"]');
    const button = wrapper.find('Button');
    console.log(button.debug());
  });
});

const textContent = node => {
  try {
    // enzyme sometimes blows up on text()
    return node.text();
  } catch (_e) {
    return "";
  }
};

// useful for finding a node by its text
export const findByText = (text, wrapper, options = {}) => {
  const comparator = options.exact
    ? x => textContent(x) === text
    : x => new RegExp(text).test(textContent(x));
  return wrapper.findWhere(comparator).last();
};
