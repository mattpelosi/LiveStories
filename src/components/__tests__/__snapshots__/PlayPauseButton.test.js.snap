// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Conditional rendering and onClick event renders pause icon when props.paused = false 1`] = `
<button
  class="MuiButtonBase-root-28 MuiButton-root-2 MuiButton-text-4 MuiButton-outlined-10 MuiButton-sizeLarge-26 PlayPauseButton-button-1"
  data-test="button"
  tabindex="0"
  type="button"
>
  <span
    class="MuiButton-label-3"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root-31"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M6 19h4V5H6v14zm8-14v14h4V5h-4z"
      >
        
      </path>
      <path
        d="M0 0h24v24H0z"
        fill="none"
      >
        
      </path>
    </svg>
  </span>
  <span
    class="MuiTouchRipple-root-38"
  >
    
  </span>
</button>
`;

exports[`Conditional rendering and onClick event renders play icon when props.paused = true 1`] = `
<button
  class="MuiButtonBase-root-28 MuiButton-root-2 MuiButton-text-4 MuiButton-outlined-10 MuiButton-sizeLarge-26 PlayPauseButton-button-1"
  data-test="button"
  tabindex="0"
  type="button"
>
  <span
    class="MuiButton-label-3"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root-31"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M8 5v14l11-7z"
      >
        
      </path>
      <path
        d="M0 0h24v24H0z"
        fill="none"
      >
        
      </path>
    </svg>
  </span>
  <span
    class="MuiTouchRipple-root-38"
  >
    
  </span>
</button>
`;
